### DISCUSSION TITLE
What is happening with dbeaver stability?

### DISCUSSION THREAD

**Discussion by dpalharini** [dbeaver/dbeaver#16229] [Q&A]
I'm been using DBeaver for the last year and a half and it was fantastic. It was the best tool I have ever used by far.

But the last 3 minor versions it became impossible to use. Every query or so I need to kill the dbeaver (with killall) and start all over again.

What happened? Is just the linux version?

**Comment by mkgrgis** [comment]
@dpalharini, have You got unstable network connection and DB disconnections? In my experience this is the biggest problem for DBeaver's stability.

**Comment by dpalharini** [comment]
@mkgrgis This behavior has been steady by the last 3 months in a roll. Sometimes, I had just opened DBeaver, connect to the database and try to see a table in a specific scheme and it breaks. The only way to get it work is by kill all processes and start all over again.

The majority of my connections is directly with the production database server of live application. Our database is a RDS Aurora compatible with mysql. When I am running a slow, cpu intensive query, by looking at the RDS monitoring console I saw times where the CPU consumption didn't change, meaning that the Beaver is not even performing the query on database, just  showing me the progress bard and counting time as if it was performing the query.

After a downgrade to version 21.3.4, the problem reduced a lot, like 4 or 5 times a day, on version 22.0.* it happened two out of three times.

**Reply by mkgrgis** [reply]
@dpalharini I have no ideas... There was some corrupted files in my experience caused to strange behaviour. You can `rm` `.eclipse` and `.cache/DBeaver`, sometimes it helps. Try to research `.local/share/DBeaverData/workspace6/.metadata ` especially log. Selectable or full deleting of `.local/share/DBeaverData/workspace6/.metadata/.plugins` can reset some unstable configurations but only if You know source of instability.

**Reply by SSSDNSY** [reply]
changed connetions timeout for DatabaseServer?

**Reply by dpalharini** [reply]
@mkgrgis I will try to do it.

@SSSDNSY my connections timeout on RDS is 1800 seconds. The timeout in the driver property is 180000 ms.

The weirdest thing is that sometimes the query is not executed, at all, the process did not reach the database, like this case:
![image](https://user-images.githubusercontent.com/5911360/164761988-5540d74a-af8e-461f-8e2f-c9550c1c120b.png)

These screenshots are from the first query executed, I opened the DBeaver and runned the query.
The process on the database did not raise. This is the CPU chart:

![image](https://user-images.githubusercontent.com/5911360/164761906-acaad983-b90c-433c-8042-02a6dbd3f15e.png)

In a posterior execution, that actually worked, this is the result:

![image](https://user-images.githubusercontent.com/5911360/164774762-3f4fd8d9-7927-4769-80ee-f1fed0b3c225.png)

**Reply by mkgrgis** [reply]
@SSSDNSY , really some connections timeout values can initiate unstable behaviour.
